okhttp-sse/src/main/kotlin/okhttp3/sse/internal/RealEventSource.kt

```diff
@@ -40,6 +44,18 @@ internal class RealEventSource(
       }
   }
 
+  private fun updateTimeout(call: Call?, duration: Duration) {
+    if (call?.timeout() is AsyncTimeout) {
```

Comment:
I'm a bit nervous about reusing `call.timeout()` here. I guess two reasons

1) it isn't clear we are the owner of it, perhaps some other component or an interceptor is using it.
2) the casting seems unsafe, but if we create it, we know the type and can set the duration ourselves.

Reply:
What do you suggest then ?

For the first point: currently (without the PR), we cancel this exact timeout (L69), so the point would be valid for the current state.
For the 2nd, the check prevent unsafe casting